docs/src/main/asciidoc/spanner.adoc

```diff
@@ -87,44 +218,50 @@ public class MyApplication {
 	  // storing new students
 	  SimpleStudent ss = new SimpleStudent();
 	  ss.name = "student1";
-	  ss.age = 99;
+	  ss.age = 99L;
 	  spannerOperations.insert(ss);
 
 	  // getting students from Spanner
 	  List<SimpleStudent> ssList =
-	    studentRepository.findTop3DistinctByIdAndNameOrAgeOrderByAgeDesc("someId", "bob", 12);
+	    studentRepository.findTop3DistinctByIdAndNameOrAgeOrderByAgeDesc("someId", "bob", 12L);
 
 	  List<SimpleStudent> ssList2 = studentRepository.fetchBySomeQuery("some argument");
 	}
 }
 
 ----
 
-The https://docs.spring.io/spring-data/commons/docs/current/reference/html/#repositories.query-methods[query methods]
-in the `StudentRepository` are generated based on the convention of their names
-except for `fetchBySomeQuery`, which is defined in the `spanner-named-queries.properties` file
-like `SimpleStudent.fetchBySomeQuery=SELECT * FROM students WHERE name = @someArg`.
+=== Query methods
 
-When defining `StudentRepository`, it is important to note that it extends the
-`PagingAndSortingRepository`, which is supported by Spring Cloud GCP Data Spanner. The first type
-argument refers to the underlying entity, and the second type parameter refers to the `@Id` type
-of that entity. Id types must correspond to one of the https://cloud.google.com/spanner/docs/data-types[supported primary key types in Spanner].
 
-=== Configuration
+==== Resolving methods by name
 
-The Spring Boot Starter for Google Cloud Data Spanner provides the following configuration options:
+In the example above, the {spring-data-commons-ref}/#repositories.query-methods[query methods]
+in `TradeRepository` are generated based on the convention of their names.
 
-|===
-| Name | Description | Optional | Default value
-| `spring.cloud.gcp.spanner.projectId` | the GCP project ID to use | Yes | provided by Spring Cloud GCP
-| `spring.cloud.gcp.spanner.instanceId` | Spanner instance to use | No |
-| `spring.cloud.gcp.spanner.database` |
-Spanner database to use | No |
-|===
+`List<Trade> findByAction(String action)` would translate to a `SELECT * FROM trades WHERE action = ?`.
+
+You can think about the little bit extreme `List<Trade> findTop3DistinctByActionAndSymbolOrTraderIdOrderBySymbolDesc(String action, String symbol, String traderId);` as
+
+[source, sql]
+----
+SELECT DISTINCT * FROM trades
+WHERE ACTION = ? AND SYMBOL = ? AND Or TRADER_ID = ?
+ORDER BY SYMBOL DESC
+LIMIT 3
+----
+
+==== Mapping SQL to repository methods using properties file
+
+
+In the example above for `List<Trade> fetchByActionNamedQuery(String action)` does not match the naming convention.
```

Comment:
Can't parse this sentence.
What doesn't match the naming convention? The example or that method? Fix punctuation accordingly.

Reply:
I think this should be
"The example above, [method signature], does not match the naming convention."

Reply:
I'll rewrite it - I was referring to the (Spring Data Query creation naming convention)[https://docs.spring.io/spring-data/data-commons/docs/current/reference/html/#repositories.query-methods.query-creation].